### ISSUE TITLE
/blobs/digest endpoint uses a POST request

### ISSUE THREAD

**Issue by AlmightyYakob** [dandi/dandi-archive#1157]
I'm not sure why POST is used here, but that endpoint is essentially a GET. If there's no good reason for this, the method should probably be updated.

**Comment by AlmightyYakob** [comment]
> @AlmightyYakob can you include a link to the relevant code in the issue description please?

Updated.

**Comment by waxlamp** [comment]
I'm not sure either, but I suppose it's because this endpoint receives parameters in the request body. In turn, if there's no good reason to be sending parameters in the body, then I agree it looks like this could be converted to a GET endpoint. (This is another instance of Chesterton's fence, so let's make sure we understand WHY this is written the way it is.)

If you convert this to a GET endpoint, I think (for the sake of "RESTful design") you may want to merge this functionality into the `GET /blobs` endpoint, with query parameters specifying the hashing algorithm and hash value. The name of this endpoint as written ("GET /blobs/digest") implies you're retrieving a digest value.

**Comment by AlmightyYakob** [comment]
> you may want to merge this functionality into the `GET /blobs` endpoint, with query parameters specifying the hashing algorithm and hash value.

What existing `GET /blobs` endpoint are you referring to?

**Comment by AlmightyYakob** [comment]
> I assumed there was a top-level `GET /blobs` endpoint. It's strange, from a design standpoint, that this `POST /blobs/digest` endpoint exists, when I would expect it to be `GET /blobs?digest=...` instead.
> 
> Could you dig in a bit and see if you can figure out why the design looks like this?

Yes, what you said is the reason I raised this issue, I don't see any reason for that to be a POST. I can look into it a bit and change it to a GET if nothing stands out.

**Comment by waxlamp** [comment]
What I meant by my latest question was, can you see if there's a reason why there is no top level `GET /blobs` endpoint (with an optional `digest` query parameter to provide what is currently done by `POST /blobs/digest`).
